quesma/quesma/router.go

```diff
@@ -401,7 +373,36 @@ func elasticsearchInsertResult(body string, statusCode int) *mux.Result {
 	}, StatusCode: statusCode}
 }
 
+func combineSourcesFromElasticWithRegistry(sourcesFromElastic *elasticsearch.Sources, schemas map[schema.TableName]schema.Schema, normalizedPattern string) {
+	sourcesFromElastic.Indices =
+		slices.DeleteFunc(sourcesFromElastic.Indices, func(i elasticsearch.Index) bool {
+			_, exists := schemas[schema.TableName(i.Name)]
+			return exists
+		})
+	sourcesFromElastic.DataStreams = slices.DeleteFunc(sourcesFromElastic.DataStreams, func(i elasticsearch.DataStream) bool {
+		_, exists := schemas[schema.TableName(i.Name)]
+		return exists
+	})
+
+	for name, currentSchema := range schemas {
+		indexName := name.AsString()
+
+		if config.MatchName(normalizedPattern, indexName) && currentSchema.ExistsInDataSource {
+			sourcesFromElastic.DataStreams = append(sourcesFromElastic.DataStreams, elasticsearch.DataStream{
+				Name:           indexName,
+				BackingIndices: []string{indexName},
+				TimestampField: `@timestamp`,
+			})
+		}
+	}
+
+}
+
 func resolveIndexResult(sources elasticsearch.Sources) *mux.Result {
+	if len(sources.Aliases) == 0 && len(sources.DataStreams) == 0 && len(sources.Indices) == 0 {
```

Comment:
No need to check `len(sources.Aliases)` - they are just pointers to existing data structures

Reply:
Writing it this way makes it more obvious to the reader that the logic is: "empty response results in 404"